eras/shelley/impl/src/Cardano/Ledger/Shelley/Tx.hs

```diff
@@ -144,7 +145,7 @@ toCBORForSizeComputation ShelleyTx {stBody, stWits, stAuxData} =
     <> encodeNullStrictMaybe encCBOR stAuxData
 
 -- | txsize computes the length of the serialised bytes (for estimations)
-sizeShelleyTxF :: forall era. EraTx era => SimpleGetter (ShelleyTx era) Integer
+sizeShelleyTxF :: forall era. EraTx era => SimpleGetter (ShelleyTx era) Word32
 sizeShelleyTxF =
   to $
     fromIntegral
```

Comment:
Ditto 
```suggestion
sizeShelleyTxF :: forall era. EraTx era => (HasCallStack, SimpleGetter (ShelleyTx era)) Word32
sizeShelleyTxF =
  to $
    errorFail $ integralToBounded @Word64 @Word32
```

Reply:
I just noticed that the length function of `LazyByteString` actually returns an `Int64`, not a `Word64`, so it should be fine to use `fromIntegral @Int64 @Word32` here since `maxBound` of `Int64` is representable with a `Word32`.